Indicators/ChandeKrollStop.cs

```diff
@@ -0,0 +1,131 @@
+
+        /// <summary>
+        /// Gets the long stop of ChandeKrollStop.
+        /// </summary>
+        public IndicatorBase<IndicatorDataPoint> LongStop { get; }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady => Samples > 0;
+
+        /// <summary>
+        /// Required period, in data points, for the indicator to be ready and fully initialized.
+        /// </summary>
+        public int WarmUpPeriod { get; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChandeKrollStop"/> class.
+        /// </summary>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(int atrPeriod, decimal atrMult, int period)
+            : this($"CKS({atrPeriod},{atrMult},{period})", atrPeriod, atrMult, period)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChandeKrollStop"/> class.
+        /// </summary>
+        /// <param name="name">The name.</param>
+        /// <param name="atrPeriod">The period over which to compute the average true range.</param>
+        /// <param name="atrMult">The ATR multiplier to be used to compute stops distance.</param>
+        /// <param name="period">The period over which to compute the max of high stop and min of low stop.</param>
+        public ChandeKrollStop(string name, int atrPeriod, decimal atrMult, int period)
+            : base(name)
+        {
+            WarmUpPeriod = 1;
+
+            _high_stop_list = new RollingWindow<decimal>(period);
+            _low_stop_list = new RollingWindow<decimal>(period);
+
+            _atr = new AverageTrueRange(atrPeriod);
+            _atrMult = atrMult;
+            _inputValues = new RollingWindow<IBaseDataBar>(atrPeriod);
+
+            LongStop = new Identity(name + "_LongStop");
+            ShortStop = new Identity(name + "_ShortStop");
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>The input is returned unmodified.</returns>
+        protected override decimal ComputeNextValue(IBaseDataBar input)
+        {
+            _atr.Update(input);
+            _inputValues.Add(input);
+
+            var highs = _inputValues.Select(input => input.High).ToList();
+            var high_stop = highs.Max() - _atr.Current.Value * _atrMult;
+            _high_stop_list.Add(high_stop);
+
+            var lows = _inputValues.Select(input => input.Low).ToList(); ;
```

Comment:
Could use Min/Max indicators avoid iterating/tolist on ever data point, same for _low_stop_list/_high_stop_list below

Reply:
Done